partition-manager/generation/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+#![no_std]
+
+#[cfg(not(target_os = "none"))]
+pub use internal::*;
```

Comment:
Question for my learning: if this crate exports nothing unless it's targetting an OS, why is it no_std? It's only used by the proc-macro crate and those don't need to bu no_std to be used by no_std crates.

Reply:
This is because the current CI setup for this repository assumes all crates to be `no_std` compatible by checking `no_std` targets. Here I had to make a choice to either change the CI such that it would leave out this library crate, or change the library crate to work for `no_std` as well.

This is a long standing discussion with the cargo team and the embedded WG to get big workspaces like `embedded-services` to support very distinct targets in a workable manner, but unfortunately we are not there yet.